services/ux-backend/handlers/expandstorage/handler.go

```diff
@@ -0,0 +1,287 @@
+	} else {
+		// create cephFilesystem
+		createCephFilesystem(w, r, client, ExpandStorage.PoolDetails.PoolName, ExpandStorage.LsoStorageClassName, mutate, ExpandStorage.PoolDetails.DataProtectionPolicy, ExpandStorage.PoolDetails.EnableCompression, ExpandStorage.PoolDetails.FilesystemName, namespace)
+
+		// create storageClass
+		createCephFilesystemStorageClass(w, r, client, ExpandStorage.StorageClassDetails.Name, ExpandStorage.PoolDetails.PoolName, ExpandStorage.StorageClassDetails.ReclaimPolicy, ExpandStorage.StorageClassDetails.VolumeBindingMode, ExpandStorage.StorageClassDetails.EnableStorageClassEncryption, ExpandStorage.PoolDetails.FilesystemName, namespace, mutate)
+	}
+
+}
+
+func updateStorageCluster(w http.ResponseWriter, r *http.Request, client client.Client, namespace string, lsoStorageClassName string, enableEncryption bool, storage, count, replica int, storageClusterName string, mutate func() error) {
+	storageCluster := &ocsv1.StorageCluster{}
+	err := client.Get(r.Context(), types.NamespacedName{Name: storageClusterName, Namespace: namespace}, storageCluster)
+	if err != nil {
+		klog.Errorf("failed to get storageCluster: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	storageQty := resource.MustParse(fmt.Sprintf("%d", storage))
+
+	deviceSet := ocsv1.StorageDeviceSet{
+		Name:      lsoStorageClassName,
+		Count:     count,
+		Replica:   replica,
+		Portable:  false,
+		Encrypted: &enableEncryption,
+		DataPVCTemplate: corev1.PersistentVolumeClaim{
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes:      []corev1.PersistentVolumeAccessMode{"ReadWriteOnce"},
+				StorageClassName: &lsoStorageClassName,
+				Resources: corev1.VolumeResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: storageQty,
+					},
+				},
+			},
+		},
+	}
+
+	storageCluster.Spec.StorageDeviceSets = append(storageCluster.Spec.StorageDeviceSets, deviceSet)
+
+	_, err = ctrl.CreateOrUpdate(r.Context(), client, storageCluster, mutate)
+	if err != nil {
+		klog.Errorf("failed to update storageCluster: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func createCephBlockPool(w http.ResponseWriter, r *http.Request, client client.Client, poolName, lsoStorageClassName string, mutate func() error, dataProtectionPolicy int, enableCompression bool, namespace string) {
+	compressiom := "none"
+	if enableCompression {
+		compressiom = "aggressive"
+	}
+	cephBlockPool := &cephv1.CephBlockPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      poolName,
+			Namespace: namespace,
+		},
+		Spec: cephv1.NamedBlockPoolSpec{
+			PoolSpec: cephv1.PoolSpec{
+				FailureDomain: "host",
```

Comment:
will FD always be "host", just to confirm that we are sure ??

asking because UI reads it from the StorageCluster CR's status and set the value accordingly, instead of hardcoding "host" always.

Reply:
We can get from UI then